### PR TITLE
Add dict configuration

### PR DESCRIPTION
**Pull request by mraspaud**
Add dict configuration to pycoast, so that configs can be passed as dictionaries and pycoast takes care of the rest.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

File: pycoast/cw_base.py [hunk 1]
```diff
@@ -803,6 +826,19 @@ def add_overlay_from_config(self, config_file, area_def):
 
         return foreground
 
+    def add_overlay_from_config(self, config_file, area_def):
+        """Create and return a transparent image adding all the overlays contained in a configuration file.
+
+        :Parameters:
```
Comment (djhoese) [inline pycoast/cw_base.py]: Is this the standard docutils/rst way of specifying parameters? It's been so long I can't remember.

File: pycoast/cw_base.py [hunk 2]
```diff
@@ -736,19 +768,10 @@ def add_overlay_from_config(self, config_file, area_def):
...
-
         is_agg = self._draw_module == "AGG"
 
-        # Coasts
-        for section, fun in zip(['coasts', 'rivers', 'borders'],
+        # Coasts, rivers, borders
+        for section, fun in zip(['coasts', 'rivers', 'borders', 'grid'],
                                 [self.add_coastlines,
                                  self.add_rivers,
                                  self.add_borders]):
```
Comment (djhoese) [inline pycoast/cw_base.py]: Doesn't add_grid need to be added here?

Comment (mraspaud) [reply pycoast/cw_base.py]: This is still WIP...

File: pycoast/cw_base.py (hunk 1 above)
Comment (mraspaud) [reply pycoast/cw_base.py]: Looks like it's working in the old functions: https://pycoast.readthedocs.io/en/latest/api/pycoast.html

**Pull request merged**